app/controllers/api/v1/matches_controller.rb

```diff
@@ -0,0 +1,15 @@
+module Api
+  module V1
+    class MatchesController < ApiController
+      def index
+        @matches = all_matches
+      end
+
+      private
+
+      def all_matches
+        current_user.first_matches.all + current_user.second_matches.all
```

Comment:
.all not needed

Reply:
Concatenating these arrays would achieve better performance. Check (this)[https://stackoverflow.com/a/23763458] out.